collector/src/main.rs

```diff
@@ -745,137 +686,196 @@ fn main() {
-            (@arg SITE_URL: +required +takes_value "Site URL")
+// For each subcommand we list the mandatory arguments in the required
+// order, followed by the options in alphabetical order.
+#[derive(Debug, clap::Subcommand)]
+#[clap(rename_all = "snake_case")]
+enum Commands {
+    /// Benchmarks a local rustc
+    BenchLocal {
+        #[clap(flatten)]
+        rustc: RustcOption,
 
-            // Options
-            (@arg DB:           --db  +takes_value "Database output file")
-            (@arg BENCH_RUSTC: --("bench-rustc")
-             "Run the special `rustc` benchmark")
-            (@arg SELF_PROFILE: --("self-profile") "Collect self-profile data")
-        )
+        #[clap(flatten)]
+        id: IdOption,
 
-        (@subcommand bench_published =>
-            (about: "Benchmarks a published toolchain for perf.rust-lang.org's dashboard")
+        #[clap(flatten)]
+        local: LocalOptions,
 
-            // Mandatory arguments
-            (@arg TOOLCHAIN: +required +takes_value "Toolchain (e.g. stable, beta, 1.26.0)")
+        #[clap(flatten)]
+        db: DbOption,
 
-            // Options
-            (@arg DB: --db +takes_value "Database output file")
-        )
+        #[clap(flatten)]
+        bench_rustc: BenchRustcOption,
 
-        (@subcommand profile_local =>
-            (about: "Profiles a local rustc with one of several profilers")
-
-            // Mandatory arguments
-            (@arg PROFILER: +required +takes_value
-             "One of: 'self-profile', 'time-passes', 'perf-record',\n\
-             'oprofile', 'cachegrind', 'callgrind', 'dhat', 'massif',\n\
-             'eprintln', 'llvm-lines', 'mono-items', 'dep-graph'")
-            (@arg RUSTC:    +required +takes_value "The path to the local rustc to benchmark")
-            (@arg ID:       +required +takes_value "Identifier to associate benchmark results with")
-
-             // Options
-            (@arg BUILDS: --builds       +takes_value
-             "One or more (comma-separated) of: 'Check', \n\
-             'Debug', 'Doc', 'Opt', 'All'")
-            (@arg CARGO:   --cargo       +takes_value "The path to the local Cargo to use")
-            (@arg EXCLUDE: --exclude     +takes_value
-             "Exclude all benchmarks that are listed in\n\
-             this comma-separated list of patterns")
-            (@arg INCLUDE: --include     +takes_value
-             "Include only benchmarks that are listed in\n\
-             this comma-separated list of patterns")
-            (@arg OUT_DIR: --("out-dir") +takes_value "Output directory")
-            (@arg RUNS:    --runs        +takes_value
-             "One or more (comma-separated) of: 'Full',\n\
-             'IncrFull', 'IncrUnchanged', 'IncrPatched', 'All'")
-            (@arg RUSTDOC: --rustdoc +takes_value "The path to the local rustdoc to benchmark")
-        )
+        /// The number of iterations to do for each benchmark
+        #[clap(long, default_value = "1")]
```

Comment:
nit: You *could* use `default_value_t = 1` here.

Reply:
Yes, but I used `default_value` for consistency with the other ones.